server/atlassian_connect.go

```diff
@@ -95,6 +96,32 @@ func httpACUninstalled(p *Plugin, w http.ResponseWriter, r *http.Request) (int,
 			errors.New("method " + r.Method + " is not allowed, must be POST")
 	}
 
+	// Notify users we have uninstalled an instance
```

Comment:
Let me clarify what @levb means here. Since this is a public route I can create a DOS attack here by spamming uninstall. So we need to at least check the request is valid before sending this event.

Reply:
@crspeller @levb I get that, and makes sense. I'll reproduce a message I sent to Lev on Friday:

I've been having trouble with JWT... when Atlassian sends the Uninstalled POST, it puts the JWT in the Header under "Authorization". Not in the body, like it does for the other requests. The problem is: I believe the Authorization header gets stripped out on redirect, so I can't access it through ngrok, so I can't test that it's working. 

This is what's sent in the body... Is anything here secure enough to use as a proxy to verify the request?
```javascript
{
    "key": "mattermost_https_chrisp_ngrok_io",
    "clientKey": "5236f67b-1021-3c6f-94da-b4e016d62ac8",
    "publicKey": "MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCvtU0OP6fwW1JsB4tfBESeObHF/tMqvJLa18NZo7wyTYQHqncE6ucJF0hzuPc/0lEOT28DvxJZRXF2WBDqRc2XRkrZnkfEnFPoQU4NLvv9DYY+/zLp0hwrDNasY0qVL5SxaBeA1PafjDqvwt+rn/bRsSyUlDyxzVYEsb7qc0hGdQIDAQAB",
    "serverVersion": "100103",
    "pluginsVersion": "1.403.0",
    "baseUrl": "https://mmtest.atlassian.net",
    "productType": "jira",
    "description": "Atlassian JIRA at https://mmtest.atlassian.net ",
    "eventType": "uninstalled"
}
```

Reply:
1/5 I suggest removing /uninstalled support if we can't easily make this work. I was talking to @jasonblais this morning about the (eventual, 2.1) need for a `/jira uninstall` command to remove the current Jira Server instance, it can also be used in lieu of `/uninstall` http callback.

For 2.0 there will be no way to properly uninstall the instance, we will recommend disabling the plugin.

Reply:
@levb For 2.0 there is a proper way to uninstall (for cloud only), and it's through this uninstall endpoint. The only problem is the JWT authorization. I'm 95% sure what I have will work, I just can't test it on my machine (and no-one using ngrok can test it either, if I'm right in assuming the problem is that the Authorization header gets stripped).

Howabout I make `/jira uninstall` work for v2.0? Maybe a separate ticket for that. @jasonblais ?

Reply:
I've created a ticket, we can discuss tomorrow during sprint planning if that goes to v2.0 or v2.1: https://mattermost.atlassian.net/browse/MM-15828

Reply:
resolving because we're going with https://mattermost.atlassian.net/browse/MM-15828